Compile error when building gni provider

When trying to build with the gni provider enabled on a Cray XC running CLE 6:

```
$ module load PrgEnv-gnu
$ ./configure CC=cc --enable-gni --enable-debug --prefix=$BUILDPATH
<...>
***
*** Built-in providers:	shm rxd rxm tcp udp gni sockets
*** DSO providers:
***
$ make
<...>
  CC       prov/gni/src/src_libfabric_la-gnix_atomic.lo
In file included from ./prov/gni/include/gnix.h:71:0,
                 from prov/gni/src/gnix_atomic.c:39:
./prov/gni/include/gnix_util.h:152:20: error: redefinition of 'dlist_remove_init'
 static inline void dlist_remove_init(struct dlist_entry *e)
                    ^~~~~~~~~~~~~~~~~
In file included from ./include/fi_mem.h:41:0,
                 from ./include/fi.h:51,
                 from ./prov/gni/include/gnix.h:59,
                 from prov/gni/src/gnix_atomic.c:39:
./include/fi_list.h:92:20: note: previous definition of 'dlist_remove_init' was here
 static inline void dlist_remove_init(struct dlist_entry *item)
                    ^~~~~~~~~~~~~~~~~
Makefile:10013: recipe for target 'prov/gni/src/src_libfabric_la-gnix_atomic.lo' failed
make[1]: *** [prov/gni/src/src_libfabric_la-gnix_atomic.lo] Error 1
make[1]: Leaving directory '<...>'
Makefile:3275: recipe for target 'all' failed
make: *** [all] Error 2
```

This seems to have been introduced by the changes in b86c958a4b615239ff7258c3a601b727455534a4.

--- a/prov/gni/include/gnix_util.h
+++ b/prov/gni/include/gnix_util.h
@@ -149,12 +149,6 @@ static inline void dlist_node_init(struct dlist_entry *e)
 	e->prev = e->next = NULL;
 }
 
-static inline void dlist_remove_init(struct dlist_entry *e)
-{
-	dlist_remove(e);
-	e->prev = e->next = e;
-}
-
 #define DLIST_IN_LIST(e) e.prev != e.next
 
 #define DLIST_HEAD(dlist)  struct dlist_entry dlist = { &(dlist), &(dlist) }

